### DISCUSSION TITLE
Ability to add folders in test suite

### DISCUSSION THREAD

**Discussion by aswingolla** [Kong/insomnia#3434] [Ideas]
**Is your feature request related to a problem? Please describe.**

Currently we don't have an option to create folder in test suite  for unit tests when we import an API spec. From the maintenance perspective this will be very helpful to segregate tests in folders based on the functionality. Imagine how it could be difficult to maintain 500+ tests running from one folder. 

If we import requests collection, then we have an option for creating the folders. In this case, we cannot add assertions like we do from Test suite.

**Describe the solution you'd like**
1. Ability to create folder and subfolders in a Test suite
2. Ability to move tests into different folders
3. Ability to duplicate folder including files in the folder
4. Ability to rename the folder

**Describe alternatives you've considered**
Didn't find any alternative so far.

**Additional context**

1. Import an API  spec file - No option to create folder in test suite shown below

<img width="1680" alt="Screen Shot 2021-05-25 at 10 18 28 pm" src="https://user-images.githubusercontent.com/30483779/119496765-2c93bd80-bda7-11eb-99a1-8b33da019f5b.png">

2. Import requests collection - Have option to create folders as shown below

<img width="1680" alt="Screen Shot 2021-05-25 at 10 36 44 pm" src="https://user-images.githubusercontent.com/30483779/119499204-d3795900-bda9-11eb-9b6b-2fbd72586c46.png">

**Comment by reynolek** [comment]
I can see this being valuable as test suits grow. The organization of the data will need to be thought through, so this is likely a fairly large change to support this.

I believe the original thought was to use `Test Suites` as the organizing process.

**Reply by aswingolla** [reply]
A Test suite generally could comprise of Feature folders -> Test Scenarios -> Test cases. It's quite easy to add many API automated tests and have a huge scope to increase the test suite in a very short time.  We could also add subfolders from different journeys perspective for ease of maintenance.

I also feel like we need a tree structure or something else for easy navigation rather than continuous scrolling (if test suites keep growing) even if we consider adding a scenario in test suite for current organizing process.

**Reply by aswingolla** [reply]
Hi Team, do you have any plan to consider this feature?

**Reply by dimitropoulos** [reply]
I would definitely welcome a PR for this.  Although I'm not making the prioritization decisions I can say that I don't know of any plan to do this very soon.  At the same time, there's hope.  We're spending more time (not less) lately working through requests like this. cc: @wdawson

**Comment by dimitropoulos** [comment]
I tried

```ts
describe('test', () => {
	it('first it', async () => {
		const response1 = await insomnia.send();
		expect(response1.status).to.equal(200);
	});
});
```

To see if it would work as a hacky workaround (but, it doesn't).

**Comment by sudo-run-dos-run** [comment]
Maybe the TEST tab should inherit the folder structure from the DEBUG tab and only offer the requests of the respective folder.
Because the "Select Request" dropdown menu for the test cases grows quite large if a project contains a large amount of requests.

I'm currently also struggling to use Insomnia as part of a test stack and although I want to love it, it's hard.

It would also be helpful to modify query params or JSON body elements directly in the TEST tab, which would allow to reuse requests and avoid bloating the DEBUG tab.
